app/web_framework/cluster_base_handler.py

```diff
@@ -62,6 +69,24 @@ def _handle_request_exception(self, ex):
         self.set_status(status_code)
         self.finish()
 
+    # For methods that annotate with `Tuple`, use `disable=invalid-sequence-index`
+    # This bug is fixed in https://github.com/PyCQA/astroid/commit/563031aaf13a44adc8db4f8d0ab8020d550aae00
+    # More information on the issue in https://github.com/PyCQA/pylint/issues/1212
+
+    def get_pagination_params(self) -> Tuple[int, int]:  # pylint: disable=invalid-sequence-index
+        offset = self.get_query_argument('offset', pagination_constants['DEFAULT_OFFSET'], True)
+        limit = self.get_query_argument('limit', pagination_constants['DEFAULT_LIMIT'], True)
+        return self._validate_arguments(int(offset), int(limit))
+
+    @staticmethod
+    def _validate_arguments(offset: int, limit: int) -> Tuple[int, int]:  # pylint: disable=invalid-sequence-index
+        offset = offset or pagination_constants['DEFAULT_OFFSET']
+        limit = limit or pagination_constants['DEFAULT_LIMIT']
+        # Negative offsets or limits are not allowed
+        non_negative_offset = max(offset, 0)
```

Comment:
1.  I'd suggest doing
```python
try:
  a = int(string_variable)
  b = int(other_string_variable)
catch ValueError as ex:
  raise BadRequestError from ex
```

For context, `ClusterBaseHandler` has a map of known exceptions to status codes. When an exception bubbles up to `ClusterBaseHandler` it calls `_handle_request_exception` (a method overwritten from tornado) and translates the exception to the relevant status_code. `BadRequestError` is one of the exceptions in this map.

2 and 3. 
I think the negative offset stuff does belong in this change. I like separating a single PR into multiple PR's when one of the following is true
(a) a single PR is too big
(b) a single PR is taking too long to implement
(c) a single PR feels like it's doing "too many things", ie 2 logically different pieces of functionality in the same commit.

I think none of those criteria apply here, so I don't see why we'd separate it. I'm not necessarily opposed to separating it, it doesn't hurt anything, I just don't see a reason to.